lib/kernel.ts

```diff
@@ -796,6 +956,52 @@ export class Kernel {
 			);
 		}
 
+		// If inserting a link, validate that there's a relationship for it
+		const contractType = contract.type.split('@')[0];
+		if (contractType === 'link') {
+			context.debug('Fetching candidate relationships for link', {
+				slug: contract.slug,
+			});
+			const linkContract: LinkContract = contract as LinkContract;
+			const unversionedFromType = linkContract.data.from.type.split('@')[0];
+			const unversionedToType = linkContract.data.to.type.split('@')[0];
+
+			const results = this.getRelationships().filter(
+				(rel) =>
+					rel.type === 'relationship@1.0.0' &&
```

Comment:
Is the type check necessary here?

Reply:
No, you're right. Leftover from when this logic was a query.